src/controllers/QuestionnaireController.ts

```diff
@@ -54,4 +54,32 @@ export class QuestionnaireController {
             }
         );
     }
+
+    /**
+     * Provide the questionnaire data for the requested questionnaire url and version
+     *
+     * @param {ISecureRequest} req
+     * @param {Response} res
+     * @memberof QuestionnaireController
+     */
+    @Get('')
+    @Middleware([AuthorizationController.checkApiUserLogin])
```

Comment:
Please change this line as I said before. Because questionnaires are protected by subjectId. I changed this to 
 @Middleware([AuthorizationController.checkStudyParticipantLogin]) and it worked fine.

Reply:
Is there a way to protect it either with checkStudyParticipantLogin or with checkApiUserLogin? Because, otherwise in the interface component, I would be forced to retrieve a subject-Id then make an additional request to make a studyparticipant login.